base/src/main/java/uk/ac/ebi/atlas/testutils/RandomDataTestUtils.java

```diff
@@ -0,0 +1,98 @@
+package uk.ac.ebi.atlas.testutils;
+
+import com.google.common.collect.ImmutableList;
+import uk.ac.ebi.atlas.experimentpage.baseline.tsne.TSnePoint;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Stream;
+
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+import static org.apache.commons.lang3.RandomStringUtils.randomNumeric;
+
+public class RandomDataTestUtils {
+    public static String getRandomExperimentAccession() {
+        // n / 456,975,543,024 chance of clashing for n experiments in the test database, let’s roll!
+        return "E-" + randomAlphabetic(4).toUpperCase() + "-" + randomNumeric(1, 6);
+    }
+
+    public static String getRandomEnsemblGeneId() {
+        return "ENS" + randomAlphabetic(4).toUpperCase() + randomNumeric(12);
```

Comment:
Shouldn't this be generating mixed and lower case too? 😉

Reply:
🤨